### PR TITLE
feat(exit-code): configure exit code on warning

### PR DESCRIPTION
**Pull request by tuxes3**
Furthermore i removed all deprecations and made the bundle only usable starting symfony `6.3`. 

That would mean we need to tag the next version `v2.0.0`. 

Issue: #1

### CODE REVIEW

**Comment by tuxes3** [comment]
@xarem these checks are getting out of hand ...

**Comment by xarem** [comment]
@tuxes3 in case of a warning, the http response code for the HTTP check is still 503. How about making it configurable like this? 

```yaml
whatwedo_monitor:
    endpoint:
        api:
            http_status_code:
                warning: 503
                critical: 503
        command:
            exit_code:
                warning: 1
                critical: 1
```

In addition, please update the docs with the change.

**Comment by tuxes3** [comment]
@xarem ready for second review

File: src/DependencyInjection/whatwedoMonitorExtension.php [hunk 1]
```diff
@@ -43,7 +42,10 @@ public function configureEndpoints(ContainerBuilder $container, array $config)
     {
         // register API
         $container->findDefinition(ApiController::class)
-            ->addArgument($config['endpoint']['api']['auth_token'] ?? null);
+            ->addArgument($config['endpoint']['api']['exit_code']['warning'] ?? 503)
+            ->addArgument($config['endpoint']['api']['exit_code']['critical'] ?? 503)
```
Comment (xarem) [inline src/DependencyInjection/whatwedoMonitorExtension.php]: @tuxes3 `http_status_code`, not `exit_code` according to the Configuration class

Comment (tuxes3) [reply src/DependencyInjection/whatwedoMonitorExtension.php]: 👀  I added a test for this case as well

**Review by xarem — APPROVED** [review]

**Pull request merged**